### PR TITLE
chore(ci,tests): add e2e test workflow to be run on after merge into …

### PR DESCRIPTION
**Pull request by odockal**
…main

### What does this PR do?
Setups individual workflow run for a E2E tests to be run on Ubuntu 22.04. Trigger offers to run the workflow on demand or once the PR is merged into main branch. 

### Screenshot/screencast of this PR
I have run workflow on my fork:
https://github.com/odockal/podman-desktop/actions/runs/6548798445/job/17784275767 

### What issues does this PR fix or reference?
#4378

### CODE REVIEW

File: .github/workflows/e2e-main.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+name: e2e-tests-main
+
+on: 
+  push:
+    branches: [main]
+    
+  workflow_dispatch:
+    inputs:
+      fork:
```
Comment (jeffmaury) [inline .github/workflows/e2e-main.yaml]: What is the purpose of these parameters ? I don't see them used in the workflow

Comment (odockal) [reply .github/workflows/e2e-main.yaml]: Nice catch! took that form different usage. will update properly.

File: .github/workflows/e2e-main.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        default: 'containers'
```
Comment (jeffmaury) [inline .github/workflows/e2e-main.yaml:27]: Someone may have a fork that may be called differently from podman-desktop so wouldn't it better to have fork be the org and the repo name (containers/podman-desktop) and use it accordingly

Comment (odockal) [reply .github/workflows/e2e-main.yaml:27]: Like this? org+repositoryName for a git repo to checkout?

**Review by jeffmaury — APPROVED** [review]
LGTM

**Comment by odockal** [comment]
@jeffmaury thx!

**Pull request merged**